build.js

```diff
@@ -994,7 +994,15 @@ var getOCamldepFlags = function(packageConfig) {
       return '';
     }
     var kind = maybeSourceKind(sourceFile, packageConfig);
-    return kind === '.ml' ? ' -impl ' + sourceFile : ' -intf ' + sourceFile;
+    // Be careful to not include e.g. .mll .mly files
+    if (kind === '.ml') {
+      return ' -impl ' + sourceFile;
+    } else if (kind === '.mli') {
+      return ' -intf ' + sourceFile;
+    } else {
```

Comment:
Sure. Maybe there should be another "kind" for lex/yacc files, and then throw if we see an unknown kind.